dev/set_matrix.py

```diff
@@ -419,6 +419,7 @@ def expand_config(config):
                         run=run,
                         package=pip_release,
                         version=ver,
+                        newer_than_max=Version(ver) > Version(max_ver),
```

Comment:
We update `ml-package-versions.yml` when we make a release like below:

```diff
# ml-package-versions.yml

sklearn:

  autologging:
    minimum: "0.20.3"
-    maximum: "0.24.1"
+    maximum: "0.24.2"
```

For this change, we want to run tests for sklearn autologging using sklearn 0.24.2 to ensure sklearn autologging really works with 0.24.2. However, `set_matrix.py` skips running tests now. Here' why:


```python
# Assumption: the latest version of sklearn is 024.2 now
# NOTE: our cross-version-tests strategy fetches the latest version available ignoring the maximum version specified in `ml-package-versions.yml`.

# ---------- NOW ----------
# Matrix generated from master
[
    {"package": "sklearn", "version": "0.24.2"},
]

# Matrix generated from PR branch
[
    {"package": "sklearn", "version": "0.24.2"},
    # -> considered as not-updated
]

# ---------- PROPOSED ----------
# Matrix generated from master (where sklearn maximum version is 0.24.1
# so `newer_than_maximum` evaluates to True)
[
    {"package": "sklearn", "version": "0.24.2", "newer_than_maximum": True},
]

# Matrix generated from PR branch (where sklearn maximum version is 0.24.2
# so `newer_than_maximum` evaluates to False)
[
    {"package": "sklearn", "version": "0.24.2", "newer_than_maximum": False},
    # -> considered as "udpated"
]

# `newer_than_maximum` field is different -> run test for sklearn 0.2

```

Reply:
This works but it's not straight forward.

Reply:
The code logic is a little strange.

Why not do:
```
# Matrix generated from master 
matrix_master = [ {"package": "sklearn", "max_supported_version": "0.24.1"}, ]

Matrix generated from PR branch:
matrix_pr = [ {"package": "sklearn", "max_supported_version": "0.24.2"}, ]
```

Then we get master and PR's max_supported_version, and then we can generate a series of CI runs. We don't need the field `newer_than_maximum` which make the logic hard to understand.

Reply:
If we did that, sklearn `0.24.2` would be always tested until  we update the max version to `0.24.2`.

Reply:
How about renaming `newer_than_max` to `supported`?